File: modules/analysis-of-algorithms/modules/worst-average-and-best-cases/src/LinearSearch.cpp
#include <bits/stdc++.h>

using namespace std;

int search(int arr[], int arrSize, int element)
{
  int i;
  for(i = 0; i < arrSize; i++)
    if(arr[i] == element)
      return i;
  return -1;
}

int main()
{
  int arr[] = {1, 10, 30, 15}; // Data.
  int element = 30; // Search element.
  int arrSize = sizeof(arr) / sizeof(arr[0]); // Array Size.

  cout << element << " is present at index " << search(arr, arrSize, element);

  return 0;
}

File: modules/analysis-of-algorithms/modules/worst-average-and-best-cases/src/LinearSearch.py
def search(arr, element):
  for index, value in enumerate(arr):
    if value == element:
      return index
  return -1

if __name__ =="__main__":
  arr = [1, 10, 30, 15]
  element = 30

  print(element, "is present at index ", search(arr, element))
